kroxylicious/src/main/templates/Kproxy/Filter.ftl

```diff
@@ -47,6 +51,10 @@ public interface ${filterClass} extends KrpcFilter {
      * @return the {@code ${msgType}} to be passed to the next filter.
      * If null is returned then the given {@code ${msgType}} will be used.
      */
-    public void on${messageSpec.name}(${dataClass} ${msgType}, KrpcFilterContext context);
+<#if messageSpec.type?lower_case == 'request'>
+    public void on${messageSpec.name}(DecodedRequestFrame<${dataClass}> ${msgType}, KrpcFilterContext context);
```

Comment:
I got the feeling `DecodedRequestFrame` might be the wrong abstraction level here. We could alternatively pass the `body` and `header` as separate parameters. We should also update the KrpcFilterContext methods to match depending on what we pick as mutating `frame.headers` and then calling `forwardRequest(body)` feels wrong.

Reply:
I agree DRF feels wrong. I like the idea of separate header and body parameters, with the obligation that the implementor forwardRequest(header, body).